drivers/entropy/entropy_rpi_pico.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024 Xudong Zheng
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <pico/rand.h>
+#include <zephyr/drivers/entropy.h>
+
+#define DT_DRV_COMPAT raspberrypi_pico_rng
+
+static int entropy_rpi_pico_get_entropy(const struct device *dev, uint8_t *buf, uint16_t len)
+{
+	__ASSERT_NO_MSG(buf != NULL);
+	uint8_t *buf_bytes = (uint8_t *)buf;
```

Comment:
Why this cast ?

Reply:
Good catch, I took that part from an earlier version of another entropy driver. The cast has been removed.